test/FunctionalTests/Microsoft.OData.Core.Tests/Build.NetFramework/SelectExpandClauseExtensionsTests.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.OData.UriParser.Aggregation;
+
+namespace Microsoft.OData.Tests
+{
+    public class SelectExpandClauseExtensionsTests
+    {
+        [Fact]
+        public void GetCurrentLevelSelectListTestMultiSelect()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("Name");
+            expected.Add("Shoe");
+
+            //Act
+            var clause = CreateSelectExpandClauseMultiSelect();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(0,expected.Except(actual).Count());
+            Assert.Equal(0,actual.Except(expected).Count());
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelect()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("Name");
+            expected.Add("FavoriteDate");
+            expected.Add("MyAddress/City");
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelect();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(0, expected.Except(actual).Count());
+            Assert.Equal(0, actual.Except(expected).Count());
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestAllSelected()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("*");
+            
+            //Act
+            var clause = CreateSelectExpandClauseAllSelected();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(expected[0], actual[0]);            
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestAllSelectedWithNamespace()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("namespace.*");
+
+            //Act
+            var clause = CreateSelectExpandClauseAllSelectedWithNamespace();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(expected[0], actual[0]);
+        }
+
+        private SelectExpandClause CreateSelectExpandClauseMultiSelect()
+        {
+            ODataSelectPath personNamePath = new ODataSelectPath(new PropertySegment(HardCodedTestModel.GetPersonNameProp()));
+            ODataSelectPath personShoePath = new ODataSelectPath(new PropertySegment(HardCodedTestModel.GetPersonShoeProp()));
+
+            var clause = new SelectExpandClause(new List<ExpandedNavigationSelectItem>(), false);

```

Comment:
>List<ExpandedNavigationSelectItem>() [](start = 52, length = 36)

why do you use "new List<ExpandedNavigationSelectItem>()", it seems you should use "new List<SelectItem>()"

Reply:
#Resolved